webapp/sources/rudder/rudder-rest/src/main/scala/com/normation/rudder/rest/lift/ComplianceApi.scala

```diff
@@ -190,6 +193,73 @@ class ComplianceApi(
     }
   }
 
+  object ExportCSV extends LiftApiModule {
+    val schema        = API.ExportDirectiveComplianceCSV
+    val restExtractor = restExtractorService
+    def process(
+        version:     ApiVersion,
+        path:        ApiPath,
+        directiveId: String,
+        req:         Req,
+        params:      DefaultParams,
+        authzToken:  AuthzToken
+    ): LiftResponse = {
+      implicit val action   = schema.name
+      implicit val prettify = params.prettify
+
+      (for {
+        level     <- restExtractor.extractComplianceLevel(req.params)
+        t1         = System.currentTimeMillis
+        precision <- restExtractor.extractPercentPrecision(req.params)
+        id        <- DirectiveId.parse(directiveId).toBox
+        t2         = System.currentTimeMillis
+
+        directive <- complianceService.getDirectiveCompliance(id, level)
+        t3         = System.currentTimeMillis
+        _          = TimingDebugLogger.trace(s"API Export compliance to CSV - getting query param in ${t2 - t1} ms")
```

Comment:
Why do you put your traces at the end ? you shoud make them just below what they are doing